helix-term/src/commands.rs

```diff
@@ -4411,7 +4410,19 @@ fn join_selections_impl(cx: &mut Context, select_space: bool) {
             end = skip_while(slice, end, |ch| matches!(ch, ' ' | '\t')).unwrap_or(end);
 
             // need to skip from start, not end
-            let change = (start, end, Some(fragment.clone()));
+            let change = {
+                let separator = {
+                    let line_end_index = line_end_char_index(&slice, line + 1);
+                    let line_contains_only_space = end == line_end_index;
+                    if line_contains_only_space {
+                        None
+                    } else {
+                        Some(Tendril::from(" "))
+                    }
+                };
+
+                (start, end, separator)
+            };
             changes.push(change);
```

Comment:
Just some style nits:

* I'd like to avoid the extra nesting here too
* Turning an `if` condition into a var with a good name is a nice tool for long `if` conditions but IMO it's always clearer to just have a comment explaining what the `if` is branching on:

```rust
let separator = if end == line_end_char_index(&slice, line + 1) {
    // The line only contains whitespace so skip adding a space.
    None
} else {
    Some(Tendril::from(" "))
};
// need to skip from start, not end
changes.push((start, end, separator));
```